zarr/util.py

```diff
@@ -75,6 +76,16 @@ def json_loads(s: Union[bytes, str]) -> Dict[str, Any]:
     return json.loads(ensure_text(s, "utf-8"))
 
 
+def _as_int_tuple(func: Callable) -> Callable:
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> Tuple[int, ...]:
+        returned_tuple = func(*args, **kwargs)
+        return tuple(int(t) for t in returned_tuple)
+
+    return wrapper
+
+
+@_as_int_tuple
```

Comment:
`normalize_shape` already calls `shape = tuple(int(s) for s in shape)`, so what is the purpose of decorating this function here?

Reply:
This is a safeguard for a future additions of early returns. I can remove the decorator here if desired.

Reply:
what about removing the decorator and adding a `chunks = tuple(int(s) for s in chunks` line to the end of normalize chunks?

Reply:
I had that as an option in commit https://github.com/zarr-developers/zarr-python/pull/1470/commits/4693cb896be755866a74434d472c421a87393ce6 but there are 3 different return statements in `normalize_chunks`.

Reply:
in that case I would recommend introducing a `_chunks` variable in that function that gets returned at the end, and replacing the early returns with assignment to `_chunks`

Reply:
Note: None of the early returns in `normalize_chunks` break my added tests

Reply:
I think I prefer option 3, because at the end of the day this is just about ensuring that functions return the correct type, and I don't think we need decorators for that.

Reply:
I reverted to option 3